internal/accounts/create.go

```diff
@@ -134,3 +147,301 @@ func create(
+		address = events.GetAddressForKeyAdded(pubKey)
+		if address != nil {
+			break
+		}
+	}
+
+	if address == nil {
+		//TODO:sideninja 200 blocks might not be enough time for the user to sign into their wallet and create the account on mainnet
+		if lastHeight-startHeight > 200 { // if something goes wrong don't keep waiting forever to avoid spamming network
+			return nil, fmt.Errorf("failed to get the account address due to time out")
+		}
+
+		time.Sleep(time.Second * 2)
+		address, err = getAccountCreatedAddressWithPubKey(service, pubKey, startHeight)
+		if err != nil {
+			return nil, err
+		}
+
+		return address, nil
+	}
+
+	return address, nil
+}
+
+func saveAccount(
+	loader flowkit.ReaderWriter,
+	state *flowkit.State,
+	account *flowkit.Account,
+	network config.Network,
+	saveToGitIgnore bool,
+) error {
+	// If using emulator, save account private key to main flow.json configuration file
+	if network == config.DefaultEmulatorNetwork() {
+		return saveAccountToMainConfigFile(state, account)
+	}
+
+	// Otherwise, save to a separate {accountName}.private.json file.
+	return saveAccountToPrivateConfigFile(loader, state, account, saveToGitIgnore)
+}
+
+func saveAccountToPrivateConfigFile(
+	loader flowkit.ReaderWriter,
+	state *flowkit.State,
+	account *flowkit.Account,
+	saveToGitIgnore bool,
+) error {
+	privateAccountFilename := fmt.Sprintf("%s.private.json", account.Name())
+	// Step 1: save the private version of the account (incl. the private key)
+	// to a separate JSON file.
+	err := savePrivateAccount(loader, privateAccountFilename, account, saveToGitIgnore)
+	if err != nil {
+		return err
+	}
+
+	// Step 2: update the main configuration file to inlcude a reference
+	// to the private account file.
+	fromFileAccount := flowkit.
+		NewAccount(account.Name()).
+		SetFromFile(privateAccountFilename)
+
+	state.Accounts().AddOrUpdate(fromFileAccount)
+
+	err = state.SaveDefault()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func saveAccountToMainConfigFile(
+	state *flowkit.State,
+	account *flowkit.Account,
+) error {
+	state.Accounts().AddOrUpdate(account)
+	err := state.SaveDefault()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func savePrivateAccount(
+	loader flowkit.ReaderWriter,
+	fileName string,
+	account *flowkit.Account,
+	saveToGitIgnore bool,
+) error {
+	account.EnableAdvancedSaveFormat()
+	privateState := flowkit.NewEmptyState(loader)
+	privateState.Accounts().AddOrUpdate(account)
+
+	err := privateState.Save(fileName)
+	if err != nil {
+		return err
+	}
+	if saveToGitIgnore {
+		addToGitIgnore(fileName)
+	}
+
+	return nil
+}
+
+func addToGitIgnore(
+	filename string,
+) error {
+	currentWd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	gitIgnoreDir := path.Join(currentWd, ".gitignore")
+
+	//opens .gitignore so that we can add {name}.private.json to .gitignore, will create .gitignore if it doesn't exist
+	file, err := os.OpenFile(gitIgnoreDir, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
For writing or loading files you should use the loader that is being passed to commands so we could have tests for this.

Reply:
addressed